AutoRest/Generators/Azure.Common/Azure.Common/AzureCodeGenerator.cs

```diff
@@ -229,6 +240,12 @@ internal static void FlattenResourceProperties(ServiceClient serviceClient)
             HashSet<string> typesToDelete = new HashSet<string>();
             foreach (var compositeType in serviceClient.ModelTypes.ToArray())
             {
+                if (compositeType.Extensions.ContainsKey(ExternalExtension) && 
```

Comment:
This is a partial check. Someone could provide "x-ms-external": false. We should also check for the value of the extension.

It would be nice to put all the three conditions in a method named "IsExternalResource"